### PR TITLE
docs: improve docstrings of storages

### PR DESCRIPTION
**Pull request by vdusek**
### Description

- Improve docstrings of storage classes.
- I also changed the list of main classes to reflect at least "somehow" the current public interface.

### Issues

- Relates: #304

### Testing

- Website was rendered locally.

### Checklist

- [x] CI passed

### CODE REVIEW

File: src/crawlee/storages/_dataset.py [hunk 1]
```diff
@@ -75,22 +75,32 @@ class ExportToKwargs(TypedDict):
...
+
+    To open a dataset, use the `open` class method by specifying an `id`, `name`, or `configuration`. If none are
+    provided, the default dataset for the current crawler run is used. Attempting to open a dataset by `id` that does
+    not exist will raise an error; however, if accessed by `name`, the dataset will be created if it doesn't already
+    exist.
 
     Usage:
-        dataset = await Dataset.open(id='my_dataset_id')
+    ```python
+    dataset = await Dataset.open(id='my_dataset_id')
```
Comment (janbuchar) [inline src/crawlee/storages/_dataset.py:102]: Not sure about this - isn't opening by name more common?

File: website/transformDocs.js [hunk 2]
```diff
@@ -79,7 +79,12 @@ const groupSort = (g1, g2) => {
 function getGroupName(object) {
     const groupPredicates = {
         'Errors': (x) => x.name.toLowerCase().includes('error'),
-        'Main Classes': (x) => ['Dataset', 'KeyValueStore', 'RequestQueue'].includes(x.name) || x.name.endsWith('Crawler'),
+        'Main Classes': (x) => [
```
Comment (janbuchar) [inline website/transformDocs.js:82]: Wow this is crazy. Not relevant to the PR, I just needed to let my feelings out.

**Review by janbuchar — APPROVED** [review]
One question, LGTM otherwise

File: src/crawlee/storages/_dataset.py (hunk 1 above)
Comment (vdusek) [reply src/crawlee/storages/_dataset.py:102]: Good point

**Review by janbuchar — APPROVED** [review]

**Comment by janbuchar** [comment]
Could you please link to the guide from https://github.com/apify/crawlee-python/releases/tag/v0.3.0 once this PR is out?

**Pull request merged**